be/src/exec/tablet_sink_index_channel.cpp

```diff
@@ -1091,11 +1091,11 @@ void NodeChannel::_try_diagnose(const std::string& error_text) {
     if (!config::enable_load_diagnose || _diagnose_closure != nullptr) {
         return;
     }
-    if (_rpc_timeout_ms <= config::load_diagnose_small_rpc_timeout_threshold_ms) {
-        int64_t count = s_small_rpc_timeout_count.fetch_add(1);
-        if (count % 20) {
-            return;
-        }
+    bool enable_profile = _rpc_timeout_ms > config::load_diagnose_rpc_timeout_profile_threshold_ms ||
+                          (s_small_rpc_timeout_profile_count.fetch_add(1) % 20 == 0);
+    bool enable_stack_trace = _rpc_timeout_ms > config::load_diagnose_rpc_timeout_stack_trace_threshold_ms;
```

Comment:
sometimes we need all BE stack traces.
for example, the timeout BE waits for replication finish and other BEs are stuck on replication.

Reply:
fixed. if stuck on waiting secondary replicas for a long time, will send a diagnose request to the primary replica to log the stack trace